RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -0,0 +1,265 @@
+using System.Collections.Immutable;
+using System.Data;
+
+namespace RubeesEat.Model.DB;
+
+public class DbBillRepository(IDbConnectionFactory connectionFactory) : IBillRepository
+{
+    public IReadOnlyList<Bill> GetAll()
+    {
+        const string commandText = "SELECT BillID, Date, Description FROM Bills ORDER BY Date DESC;";
+        var bills = new List<Bill>();
+
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = commandText;
+        command.CommandType = CommandType.Text;
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+
+        while (reader.Read())
+        {
+            var guid = reader.GetGuid(0);
+            bills.Add(new Bill
+            (
+                guid,
+                reader.GetDateTime(1),
+                reader.GetString(2),
+                [..GetEntryLines(guid)]
+            ));
+        }
```

Comment:
Put the variable bills in line 11 to line 23, where it is actually used, for better readability